pylhc/constants/calibration.py

```diff
@@ -0,0 +1,133 @@
+BPMS = {1: {1: ['BPMR.5L1.B1',
+                'BPMYA.4L1.B1',
+                'BPMWB.4L1.B1',
+                'BPMSY.4L1.B1',
+                'BPMS.2L1.B1',
+                'BPMSW.1L1.B1',
+                'BPMSW.1R1.B1',
+                'BPMS.2R1.B1',
+                'BPMSY.4R1.B1',
+                'BPMWB.4R1.B1',
+                'BPMYA.4R1.B1'],
+            2: ['BPM.5L1.B2', 
+                'BPMYA.4L1.B2', 
+                'BPMWB.4L1.B2', 
+                'BPMSY.4L1.B2', 
+                'BPMS.2L1.B2', 
+                'BPMSW.1L1.B2', 
+                'BPMSW.1R1.B2', 
+                'BPMS.2R1.B2', 
+                'BPMSY.4R1.B2', 
+                'BPMWB.4R1.B2', 
+                'BPMYA.4R1.B2']
+            },
+        4: {1: [
+                'BPMYA.5L4.B1',
+                'BPMWI.A5L4.B1',
+                'BPMWA.B5L4.B1',
+                'BPMWA.A5L4.B1',
+                'BPMWA.A5R4.B1',
+                'BPMWA.B5R4.B1',
+                'BPMYB.5R4.B1',
+                'BPMYA.6R4.B1',
+                ],
+            2: [ 
+                'BPMYB.5L4.B2',
+                'BPMWA.B5L4.B2',
+                'BPMWA.A5L4.B2',
+                'BPMWA.A5R4.B2',
+                'BPMWA.B5R4.B2',
+                'BPMWI.A5R4.B2',
+                'BPMYA.5R4.B2',
+                'BPMYB.6R4.B2'
+                ]
+             },
+        5: {1: ['BPMYA.4L5.B1', 
+                'BPMWB.4L5.B1', 
+                'BPMSY.4L5.B1', 
+                'BPMS.2L5.B1', 
+                'BPMSW.1L5.B1', 
+                'BPMSW.1R5.B1', 
+                'BPMS.2R5.B1', 
+                'BPMSY.4R5.B1',  #WHY
+                'BPMWB.4R5.B1', 
+                'BPMYA.4R5.B1', 
+                'BPM.5R5.B1'],
+            2: ['BPMYA.4L5.B2', 
+                'BPMWB.4L5.B2', 
+                'BPMSY.4L5.B2', 
+                'BPMS.2L5.B2', 
+                'BPMSW.1L5.B2', 
+                'BPMSW.1R5.B2', 
+                'BPMS.2R5.B2', 
+                'BPMSY.4R5.B2', 
+                'BPMWB.4R5.B2', 
+                'BPMYA.4R5.B2', 
+                'BPMR.5R5.B2']
+            }
+        }
+
+# For the dispersion method, only a subject of the BPMs is used
+# Same as BPM: IP and then beam
+D_BPMS = {1: {1: ['BPMSY.4L1.B1',
+                  'BPMS.2L1.B1',
+                  'BPMSW.1L1.B1',
+                  'BPMSW.1R1.B1',
+                  'BPMS.2R1.B1',
+                  'BPMSY.4R1.B1'],
+              2: ['BPMSY.4L1.B2', 
+                  'BPMS.2L1.B2', 
+                  'BPMSW.1L1.B2', 
+                  'BPMSW.1R1.B2', 
+                  'BPMS.2R1.B2', 
+                  'BPMSY.4R1.B2']
+              },
+          4: {1: [],
+              2: []
+              },
+          5: {1: ['BPMSY.4L5.B1', 
+                  'BPMS.2L5.B1', 
+                  'BPMSW.1L5.B1', 
+                  'BPMSW.1R5.B1', 
+                  'BPMS.2R5.B1', 
+                  #'BPMSY.4R5.B1', WHY
```

Comment:
why is this one commented out? Was also not used in the prev code or results in errors when executing?

Reply:
It wasn't used before and it's not in the dispersion TFS file, nobody seems to know why.
Should I investigate more?

Reply:
does the code still work if not all BPMs are found? then I'd uncomment it and if it's not in the dispersion file nothing should change in your prev. results ... and if the code can't handle missing BPMs, I'd add an info message at some point that not all BPMs were found in the measurement and fit & calibration might not be as good as can be .. leaves the obligation with the missing BPM with the (unfortunate) user

Reply:
I approve. Should probably make the logging message be `warning` level too

Reply:
The code now works with missing BPMs.
The resulting TFS files will have NaN values for those BPMs and they won't be used for the curve fit.
A warning is displayed via the logger.